src/backend/distributed/utils/maintenanced.c

```diff
@@ -257,6 +263,13 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		CHECK_FOR_INTERRUPTS();
 
 		/*
+		 * We clear the metadata cache before every iteration because otherwise
+		 * it might contain stale OIDs if deadlock detection runs during a DROP
+		 * EXTENSION followed by a CREATE EXTENSION.
+		 */
+		ClearMetadataCache();
```

Comment:
I think this is acceptable given that we're clearing a small amount of memory and doing two catalog reads (i.e., `pg_dist_node` and `pg_dist_logcal_group`) back to fill the required metadata.

My only comment would be that the function name is a bit strong, when I first saw it, I expected that we'd clear the distributed table caches as well.  However, I think that's not related to this PR, because the name of the struct we're clearing is `MetadataCacheData`. Probably not the best alternative, but maybe `ClearCatalogMetadataCache()` or so? (feel free to ignore this)